tools/Replay.hpp

```diff
@@ -108,6 +154,68 @@ struct Stats {
     std::unordered_map<std::string, MessageStat> _recvd;
     std::unordered_map<std::string, MessageStat> _sent;
 
+    CPUStat GetProcessTimes() {
+
+        std::ifstream fileStat("/proc/" + std::to_string(getpid()) + "/stat");
+        std::ifstream fileUpTime("/proc/uptime");
+
+        std::string statLine, upTimeLine;
+        std::getline(fileStat, statLine);
+        std::getline(fileUpTime, upTimeLine);
+
+        std::stringstream ss(statLine), us(upTimeLine);
+
+        std::vector<std::string> vStat;
+        std::string token;
+        while (ss >> token) {
+            vStat.push_back(token);
+        }
+
+        std::vector<std::string> uStat;
+        while (us >> token) {
+            uStat.push_back(token);
+        }
+
+        size_t userTime = std::stoull(vStat[13]);
+        size_t kernelTime = std::stoull(vStat[14]);
+        double startTime = std::stod(vStat[21]);
+        double upTime = std::stod(uStat[0]);
+
+        long clockTick = sysconf(_SC_CLK_TCK);
+
+        double userProcessTimeInSeconds = userTime / static_cast<double>(clockTick);
+        double kernelProcessTimeInSeconds = kernelTime / static_cast<double>(clockTick);
+        double elaspedTime = upTime - (startTime / clockTick);
+
+        CPUStat cpuStat = {
+            userProcessTimeInSeconds,
+            kernelProcessTimeInSeconds,
+            elaspedTime
+        };
+
+        return cpuStat;
+    }
+
+    size_t GetMemoryUsage()
```

Comment:
Fair enough I guess - it is amazing that this jitters so much though [!].

Reply:
I really love numbers that don't jitter. Can we addup 'Dirty' instead of PSS - which may be messed about by other processes starting/stopping that share things, and see if that helps jitter ? =)